app/api/services/ocr/OcrManager.ts

```diff
@@ -0,0 +1,233 @@
+import { Readable } from 'stream';
+import urljoin from 'url-join';
+import { files, uploadsPath, readFile } from 'api/files';
+import { generateFileName, fileFromReadStream } from 'api/files/filesystem';
+import { processDocument } from 'api/files/processDocument';
+import settings from 'api/settings/settings';
+import { TaskManager, ResultsMessage } from 'api/services/tasksmanager/TaskManager';
+import { emitToTenant } from 'api/socketio/setupSockets';
+import { tenants } from 'api/tenants/tenantContext';
+import createError from 'api/utils/Error';
+import request from 'shared/JSONRequest';
+import { FileType } from 'shared/types/fileType';
+import relationships from 'api/relationships';
+import { handleError } from 'api/utils/handleError';
+import { OcrRecord, OcrStatus } from './ocrModel';
+import { EnforcedWithId } from '../../odm/model';
+import {
+  createForFile,
+  getForSourceFile,
+  getForSourceOrTargetFile,
+  markError,
+  markReady,
+} from './ocrRecords';
+
+interface OcrSettings {
+  url: string;
+}
+
+const LANGUAGES_MAP: { [key: string]: string } = {
+  arb: 'ar',
+  deu: 'de',
+  eng: 'en',
+  fra: 'fr',
+  spa: 'es',
+};
```

Comment:
Please, do not use a locally defined hashmap!
We have `app/shared/languagesList.ts` for this exact purpose.

Reply:
Working on it. For some reason we thought it was under the frontend package.